Source/KSCrash/Reporting/Tools/KSHTTPMultipartPostBody.m

```diff
@@ -161,28 +163,34 @@ - (NSData*) data
     }
 
     NSMutableData* data = [NSMutableData dataWithCapacity:baseSize];
+    BOOL firstFieldSent = NO;
     for(KSHTTPPostField* field in _fields)
     {
-        [data appendUTF8Format:@"--%@\r\n", g_boundary];
+        if (firstFieldSent) {
+            [data appendUTF8Format:@"\r\n"];
+        } else {
+            firstFieldSent = YES;
+        }
+        [data appendUTF8Format:@"--%@\r\n", _boundary];
         if(field.filename != nil)
         {
             [data appendUTF8Format:@"Content-Disposition: form-data; name=\"%@\"; filename=\"%@\"\r\n",
-             [field.name URLEncoded],
-             [field.filename URLEncoded]];
+             field.name,
+             field.filename];
```

Comment:
These still need to be URL encoded since they may contain control characters. `appendUTF8Format` only appends he UTF-8 binary representation of whatever string you pass it.